test-data/unit/errorstream.test

```diff
@@ -0,0 +1,46 @@
+-- Test cases for incremental error streaming. Each test case consists of two
+-- sections.
+-- The first section contains [case NAME] followed by the input code, while
+-- the second section contains [out] followed by the output from the checker.
+-- Each time errors are reported, '==== Errors flushed ====' is printed.
+-- The log.log_checking() function will immediately emit a message from
+-- a plugin when a call to it is checked, which can be used to verify that
+-- error messages are printed before doing later typechecking work.
+--
+-- The input file name in errors is "file".
+--
+-- Comments starting with "--" in this file will be ignored, except for lines
+-- starting with "----" that are not ignored. The first two dashes of these
+-- lines are interpreted as escapes and removed.
+
+[case testErrorStream]
```

Comment:
Add test case with an import cycle? You can perhaps use "deferred nodes" to interleave messages from two modules (like 'error from module a', 'error from module b', 'error from module a') and then you can test that the errors are grouped by file correctly.